src/service/extensions-impl.js

```diff
@@ -168,12 +176,19 @@ export class Extensions {
   /**
    * Waits for the previously included extension to complete
    * loading/registration.
+   * @param {!Window} win
    * @param {string} extensionId
-   * @return {!Promise<!ExtensionDef>}
+   * @param {number=} opt_timeout
+   * @return {!Promise<?ExtensionDef>}
    */
-  waitForExtension(extensionId) {
-    return this.waitFor_(this.getExtensionHolder_(
-        extensionId, /* auto */ false));
+  waitForExtension(win, extensionId, opt_timeout) {
+    // Note we can't use timeout service becuase it's not instantiated yet.
+    const timer = new Timer(win);
```

Comment:
`Services.timerFor(this.win)`

Reply:
calling Services.timerFor causes a circular reference

Reply:
figured out how to do this - done